File: Java/src/session3/td/LastOccurence.java
package session3.td;

public class LastOccurence implements Visiteur<Integer>{

	private int lastOccurence;
	private char charRecherche;
	private int cpt = 0;


	public LastOccurence(char charRecherche) {
		this.charRecherche = charRecherche;
		cpt = 0;
	}

	@Override
	public Integer casVide() {
		return lastOccurence;
	}

	@Override
	public Integer casCons(char n, Integer r) {
		
		if(n==charRecherche) {
			lastOccurence = cpt;
		}
		cpt++;
		return lastOccurence;
	}

}

File: TD3/typescript/session3/td/LastOccurence.ts
import { Visiteur } from "./Visiteur";

export class LastOccurence implements Visiteur<number>{

    public lastOccurence:number;
    public charRecherche:string;
    public cpt=0;

    public constructor(charRecherche:string){
        this.charRecherche=charRecherche;
        this.cpt=0;
    }

    casVide(): number {
        return this.lastOccurence;
    }

    casCons(n:string, r:number):number{
        if(n==this.charRecherche) {
			this.lastOccurence = this.cpt;
		}
		this.cpt++;
    return this.lastOccurence;
    }


}